esphome/components/esp32/__init__.py

```diff
@@ -281,7 +281,7 @@ def _format_framework_espidf_version(
     cv.Version(5, 3, 2),
     cv.Version(5, 3, 1),
     cv.Version(5, 3, 0),
-    cv.Version(5, 1, 5),
+    cv.Version(5, 1, 6),
```

Comment:
This should not be replaced here but rather appended to the list; it's used for version validation only.
```suggestion
    cv.Version(5, 1, 5),
    cv.Version(5, 1, 6),
```